dev/system-test/firestore.ts

```diff
@@ -2321,3 +2324,271 @@ describe('Client initialization', () => {
     });
   }
 });
+
+describe('Bundle building', () => {
+  // Helper function for testing: converts "13{..}43{..}51{..}" to [[13,43,51],[obj,obj,obj]]
+  function bundleToLengthsAndObject(bundle: string): [number[], object[]] {
```

Comment:
To make this easier to digest, consider returning an array with objects: `Array<{length: number, content: IBundleElement}>`